### ISSUE TITLE
Reload-all makes purge-deleted articles re-appear

### ISSUE THREAD

**Issue by giacecco** [akrennmair/newsbeuter#329]
Hi All, 
I'm using the current [head](https://github.com/akrennmair/newsbeuter/commit/34aaaaf25910024c0337cb6c3c3f6056595a71e3). The issue is all in the subject, it looks like a bug to me, but perhaps it's a feature?

To reproduce:

1) delete (D) a few message in one feed
2) purge-delete ($)
3) go back to the feeds list (Q)
4) reload all (R)
5) go back to the feed

... and you will see the articles you deleted.

BUT, if you quit and re-start newsbeuter between (3) and (4), the problem is not there. 

In other words, it is like purge-deleted only really runs on quitting, and deleted messages are just hidden while waiting.  

If needed, my newsbeuter configuraiton is [here](https://github.com/Digital-Contraptions-Imaginarium/newsbeuter-configuration/tree/15587d19226e7eb535db30c37046298b6cfa73f1).

Thanks!

**Comment by Minoru** [comment]
Looks like a bug to me. Didn't actually check if I can reproduce it, but looks plausible from what I saw in the code.

Not yet ready to tell when I'll fix this one, though.

**Comment by giacecco** [comment]
In the meantime I've found an even more obvious way to reproduce:

1) delete (D) a few messages in one feed
2) purge-delete ($)
3) go back to the feeds list (Q)
4) go back to the feed

**Comment by tsipinakis** [comment]
I can also confirm this, it seems to happen only when deleting items from search results or query feeds but not when deleting items directly from their feeds. (At least I haven't been able to reproduce it with normal feeds.)

**Comment by giacecco** [comment]
Great @tsipinakis , that is why it always happens to me, as I read all my content through queries. @Minoru what do you reckon?

**Comment by Minoru** [comment]
I still haven't looked into this, and still don't know when I will, @giacecco.

**Comment by qoxxxx** [comment]
Can confirm, except giacecco's workaround (quit & reload) doesn't work for me, so impossible to delete articles :( 
On both 2.9 and 2.10.
